### ISSUE TITLE
Update all non-BEM components to use BEM

### ISSUE THREAD

**Issue by danielgolden** [Masterworks/Clarity-Components#12]
If we're are going to style this site using BEM we need to do so consistently. Be sure to update all of the existing components (for example, on the ui guide page) to use BEM. Once this is done, be sure to apply update those components across all of the existing pages.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden I have updated all of the existing components provided by you to the present style format, please check and let me know your view.

**Comment by danielgolden** [comment]
@Chitradip-Capitalnumbers What I meant by "update all non-bem components to use bem" was that all of the components in this project need to use the BEM naming-convention, not that they all need to use the same Sass mixins and variables (though it is true that they should all use the same system of mixins and variables).

What I was asking for was that you take the classes in a given component and adjust the class names in each to use the BEM naming convention as appropriate. Here is an example:

```html
<div class="c-select-small">
    <div class="input-wrap">
        <select class="input-box" name="input-box">
            <option value="option">option</option>
        </select>
        <label class="input-label">Label</label>
    </div>
</div>
```

should become...

```html
<div class="c-select--small">
    <div class="c-select--small__input-wrap">
        <select class="c-select--small__input-box" name="input-box">
            <option value="option">option</option>
        </select>
        <label class="c-select--small__input-label">Label</label>
    </div>
</div>
```
And then the necessary changes in the .scss files should be made. 

---
If any of this is unclear please ask any questions you have, and in the future if any of the tasks you received from us are unclear to you please do ask about them before proceeding.

**Comment by mdfrederick** [comment]
@Chitradip-Capitalnumbers As discussed, please duplicate these and change naming conventions to BEM.

Then we will integrate/replace in VueJS app, once completed in the App we will remove the old-style.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
I have started working on this. let you know the status on tomorrow.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
I have updated the button set on ui-guide-v1.html. Also, I have converted the button icon into SVG code so that we can update the color of this icon when needed. Please check and let me know the I am on the right track or not.

Thanks!

**Comment by danielgolden** [comment]
The svg change sounds good. But I'm not quite sure being on the right track. On the right track toward what?

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
Actually, I have updated the structure, I am hoping that the design remains the same. Please check progress on this file - ui-guide-v1.html

**Comment by danielgolden** [comment]
Yeah, the design looks correct with the exception of the icons of the bordered buttons because they are white against a white background you can't see them. They should be the same color as the rest of the button. Do you know what I mean?

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
Please test those on a localhost. Also, I have updated the login page in BEM. It has not completed yet. Some elements are left. I will continue this on Monday and finish this move with the other pages.
Please check this file : login-v2.html

Thanks!

**Comment by danielgolden** [comment]
Why do even the buttons that don't have an icon have the class `u-btn-outline-icon`? Also, why are these utility classes if they don't modify an object or component? Utility classes shouldn't be tied to any specific peice of UI.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
I have updated those class and maintained this throughout the project.

Thanks for your advice.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
I have pushed my code. Please check this files - edit-client-v2.html and dashboard-v1.html. Need some fine tuning. I will do those on tomorrow.

**Comment by danielgolden** [comment]
Not sure how much is covered by/included in "Needs some fine tuning", but I know that at least the inputs are not ready to go (see the code example in [my comment above](https://github.com/Masterworks/Clarity-Components/issues/12#issuecomment-405293008)). Many such issues similar to this. Let me know when you're finished with these pages and I can provide more in depth feedback.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
Below pages are converted to BEM
i. edit-client-v2 
ii. dashboard-v1
iii. edit-permission-v2
iv. edit-project-v2
v. edit-project-v2
vi. edit-role-blank-v2
Please check them and let me know you views.

**Comment by danielgolden** [comment]
> ii. dashboard-v1

- What is the significance of this class name:

**Comment by danielgolden** [comment]
That was an accident. Didn't intend to close the issue.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

I have completed request-screen-in-progress-v2. Please check that. Also fixed this issue - dashboard-v1: What is the significance of this class name: c-service__report?

v. edit-project-v2 is listed twice in your comment above. Is there a page that you meant to include, but didn't?
> Somehow I have pasted that two times. I am really sorry for that silly. Please ignore that.

I will check rest of the points on tomorrow.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

I have checked this issue. 

- This class name is redundant: `h1`. The element is already called `<h1>`. Please resolve on all pages where this exists.
- The `h1--user` class is included on many pages that are not actually related to users. We need a more generic class name to use here. Also, I don't the class name needs to be a modifier, it can just be a basic generic class name. Please resolve on all pages where this exists.

> If I remove this right now or update that then the existing pages will have some design issue. I will do this when all element has been completed. Please give me some time.

**Comment by danielgolden** [comment]
> If I remove this right now or update that then the existing pages will have some design issue. I will do this when all element has been completed. Please give me some time.

Sounds good 👍🏽

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

I have updated more pages. 
They are : 
1. submit-confirmation-v2
2. single-report-v2
3. select-client-v2
4. select-client-v2
5. projects-list-v2
6. projects-grid-v2

Please review those and let me know.

Thanks
Chitradip

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden and @mdfrederick 

Today’s work update
---------------------------
1. Added Report listing page
2. added report blank page
3. converted list-user-v2 to BEM

Please review those.

**Comment by danielgolden** [comment]
### submit-confirmation-v2.html

- Not sufficient spacing between `.c-modal__body` and `.c-modal__footer`
- `.c-btn-outline` should have a transparent background across the board (everywhere, not just on this page). Because of that the grey background of the `.c-modal__footer` should be showing through.
- `.c-modal` should have a box-shadow. See figma.

### single-report-v2.hmtl

- The "View all reports" button should be small outlined button, not the standard size.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

Today I have added 
view-all-grid-v2.html and view-all-list-v2
Also, updated those points you mention above. 
 
Now only the header tags need to be updated. After that BEM conversion is ready. 
Please review all and let me know your thoughts.

Thanks
Chitradip

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

Today I have completed 
add-user-v2
add-report 

Please check and let me know your valuable feedback.

Thanks
Chitradip
